client/cypress/integration/query/parameter_spec.js

```diff
@@ -120,6 +120,15 @@ describe("Parameter", () => {
       cy.getByTestId("TableVisualization").should("contain", "value2");
     });
 
+    it("supports filters", () => {
+      cy.getByTestId("ParameterValueInput").type("value1");
+
+      // only the filtered option should be on the DOM
+      cy.get(".ant-select-item-option").each($option => {
+        expect($option).to.contain("value1");
+      });
+    });
```

Comment:
I still think this test is small enough that it would better fit as an assertion in one of the other existing specs ([Creating “tiny” tests with a single assertion](https://docs.cypress.io/guides/references/best-practices.html#Creating-%E2%80%9Ctiny%E2%80%9D-tests-with-a-single-assertion))

You can actually create a function (`openAntdDropdownAndSearch` or sth) and apply it to tests on both the Enum and Query parameter.

Reply:
It would even be better if we updated `cy.selectAntdOption()` to support filtering

Reply:
I don't see any particular advantage on having it in both of them, as they use the same components. I made a helper function to run the assertion, but particularly thought it made the test less comprehensible, because at the end of the day, testing the filters is not really related to the other assertions. That was actually my first approach before I decided moving it to its own assertion.

Reply:
**Advantage on having it in both of them:** they have different components backing them, so it's possible for it to stop working in only one parameter type. Since we won't be constantly checking that it still works, why not add a `~100ms` assertion to cover it in existing specs?

> testing the filters is not really related to the other assertions

Check the link I sent above.

If you are worried about comprehension, a function should be enough to do so. The bigger problem is that by doing this you switch an assertion that would take those 100ms into a test that will take the whole spec setup + browser load time. BTW, we have tests that will check that you can select a dropdown value and that will work for the query, adding a search before that selection is related, no?